Makefile

```diff
@@ -3,20 +3,37 @@ BINPATH ?= build
 BUILD_TIME=$(shell date +%s)
 GIT_COMMIT=$(shell git rev-parse HEAD)
 VERSION ?= $(shell git tag --points-at HEAD | grep ^v | head -n 1)
+CORE_ASSETS_PATH=$(shell go list -f '{{.Dir}}' -m github.com/rav-pradhan/test-modules/render)
 
 .PHONY: audit
 audit:
 	go list -m all | nancy sleuth
 
 .PHONY: build
-build:
+build: generate-prod
 	go build -tags 'production' -o $(BINPATH)/dp-frontend-cookie-controller -ldflags "-X main.BuildTime=$(BUILD_TIME) -X main.GitCommit=$(GIT_COMMIT) -X main.Version=$(VERSION)"
 
 .PHONY: debug
-debug:
+debug: generate-debug
 	go build -tags 'debug' -o $(BINPATH)/dp-frontend-cookie-controller -ldflags "-X main.BuildTime=$(BUILD_TIME) -X main.GitCommit=$(GIT_COMMIT) -X main.Version=$(VERSION)"
 	HUMAN_LOG=1 DEBUG=1 $(BINPATH)/dp-frontend-cookie-controller
 
 .PHONY: test
-test:
+test: generate-debug
 	go test -race -cover ./...
+
+.PHONY: generate-debug
+generate-debug:
+	# fetch the renderer library and build the dev version
+	go get github.com/rav-pradhan/test-modules/render
```

Comment:
As I see it, we need to fetch the renderer library first in order to build the assets data file.

We run `go-bindata` with the `CORE_ASSETS_PATH` included, in order to build both app-specific and core assets into a single `data.go` output.

The `-prefix` flag removes `${CORE_ASSETS_PATH}/assets` from the core assets that are pulled in from the library. This means that the core asset names in data.go can be referenced in our code without requiring the full path - (i.e. `main.tmpl` instead of `/path/to/main.tmpl`)